packages/SwingSet/src/kernel/kernel.js

```diff
@@ -283,24 +283,41 @@ export default function buildKernel(kernelEndowments, kernelOptions = {}) {
     }
   }
 
-  function removeVatManager(vatID, reason) {
-    const old = ephemeral.vats.get(vatID);
-    ephemeral.vats.delete(vatID);
-    const err = reason ? Error(reason) : undefined;
-    old.notifyTermination(err); // XXX TODO: most of the places where notifyTermination gets passed don't need it
-    return old.manager.shutdown();
+  function rejectOldPromises(vatID, reason) {
```

Comment:
`rejectOldPromises` is not a good name; their age is not at issue.  Maybe `rejectOrphanedPromises` ?